packages/nimble-components/src/table/specs/table-column-specs/table-column-mapping.md

```diff
@@ -0,0 +1,341 @@
+
+### Angular integration
+
+Angular directives will be created for the column components and the mapping components. No component has form association, so a `ControlValueAccessor` will not be created.
+
+### Blazor integration
+
+Blazor wrappers will be created for the components. Columns will be generic in the type of the key, and will cascade that type parameter to contained mapping elements (see [`CascadingTypeParameter`](https://learn.microsoft.com/en-us/aspnet/core/blazor/components/generic-type-support?view=aspnetcore-7.0#cascaded-generic-type-support)):
+
+```HTML
+<NimbleTableColumnMapping TKey=int Field="NumberData">
+    <NimbleMappingText Key="1" Label="foo"></NimbleMappingText>
+    <NimbleMappingText Key="2" Label="bar"></NimbleMappingText>
+</NimbleTableColumnMapping>
+```
+
+### Visual Appearance
+
+The cell view (and group header view) will be responsible for styling the templates returned by the mappings. This will include alignment and spacing (`--ni-nimble-small-padding`). Unfortunately, because the mappings cannot provide CSS to go with the templates they return, some implementation knowledge will leak from the mappings to the cell/group header views. For example, we must set `flex-shrink: 0` on all elements so that icons do not get smaller in a group header when the adjacent label takes up all the space.
+
+---
+
+## Implementation
+
+### States
+
+N/A
+
+### Accessibility
+
+Text, icons, and spinner are not interactive and cannot receive keyboard focus. These items already have proper accessibility roles, and we will set accessible names (`aria-label`) based on the `label` value provided by the client.
+
+### Globalization
+
+All text will be provided by the client and is expected to be localized.
+
+### Security
+
+N/A
+
+### Performance
+
+N/A
+
+### Dependencies
+
+None
+
+### Test Plan
+
+-   Unit tests will be written verifying the usual component expectations, plus:
+    -   renders mapping matching the cell value (string, number, and boolean)
+    -   nothing rendered when value matches no mappings
+    -   validation error when non-unique mapping keys exist
+    -   validation error when multiple mappings marked as default
+    -   validation error when mapping key is null and not marked default
+    -   validation error when invalid icon name given
+    -   validation error when icon column has a `nimble-mapping-text` element
+    -   grouping header for icon column includes label
+-   Verify manually that the column content appears in the accessibility tree and can be read by a screen reader.
+-   Verify manually that several mapping columns with thousands of elements scrolls performantly.
+-   Visual Chromatic tests will be created
+
+### Tooling
+
+N/A
+
+### Documentation
+
+Documented in Storybook
+
+---
+
+## Open Issues
```

Comment:
I thought there were still some open questions on the sizing modes we thought the icon-only column needed. Is that not the case? Are we set on having it only support fixed pixel widths?

Reply:
As far as I know (and the spec says), the icon column will be fixed width and the mapping column will use either fractional width or fixed width. No one else has commented about it, so I take that to mean we're good with that. Do you think we need to do something different?